packages/calypso-e2e/src/lib/pages/plans-page.ts

```diff
@@ -81,7 +88,19 @@ export class PlansPage {
 	 * @param {PlansPageTab} targetTab Name of the navigation tab to click on
 	 */
 	async clickTab( targetTab: PlansPageTab ): Promise< void > {
-		await this.page.waitForLoadState( 'networkidle' );
+		if ( targetTab === 'My Plan' ) {
+			// Current view is Plans, going to My Plans.
+			// Wait for the plans overview to render fully.
+			const locator = this.page.locator( selectors.plansView );
+			await locator.waitFor();
+		}
+		if ( targetTab === 'Plans' ) {
+			// Current view is My Plans, going to Plans.
+			// Wait for the details of current plan to
+			// render fully.
+			const placeholderLocator = this.page.locator( `.my-plan-card${ selectors.placeholder }` );
+			await placeholderLocator.waitFor( { state: 'hidden' } );
+		}
```

Comment:
This may be slightly confusing as this takes the form of pre-processing prior to the click action on the intended tab, instead of the easier to understand post-processing after the tab has been clicked.

Because the whole reason why this is necessary is to prevent the async loading elements from swallowing up the `clickTab` action, I chose to go with the pre-processing as it would guarantee that the page state prior to the click is fully loaded (incl. async elements) and is ready to accept actions.

Reply:
I think this makes a lot of sense!

My only concern: is there a different way we can gauge the starting spot? The method is currently written to guarantee the ending tab, but there's not technically a guarantee that we're starting on the other tab (people may call this not knowing what the default tab is they will land on).

Maybe we can check the URL? That seems pretty reliable!

Reply:
If I turn your question on the head a bit; why do we need to be mindful of the starting page, if the end state is guaranteed? 

This solution _should not_ get hung up if the source and destination tabs are the same, as I do not rely on network requests but rather that placeholders are all loaded.

Reply:
Ah, I didn't explain myself well, sorry! Those are really good points.

The one, specific concern I had was... If someone is on the 'My Plan' tab, and they call this function targeting the same 'My Plan' tab, won't it throw an error because...
```typescript
const locator = this.page.locator( selectors.plansView );
await locator.waitFor();
```
... won't match on anything? Or am I misunderstanding?

Reply:
Ah okay, I think I get what you mean.

I did try out duplicating a call like this:

```
			await plansPage.clickTab( 'My Plan' );
			await plansPage.clickTab( 'My Plan' );
```

and I was able to cause it to hang, which should be the case you're referring to here.

I think maybe I'll try reviewing how this is done. There must be a better way to do this that doesn't rely on knowing where the originating tab is.

Reply:
I've pushed a much better solution in 60756d6cb549d76ae29c1a1647b680b562f3120f. I'd like to hear your thoughts.

Reply:
Ahh yes, that short circuit approach is nice! LGMT! 🚀